### PR TITLE
Update secret handler for PUT method

### PR DESCRIPTION
**Pull request by viveksyngh**
This commit updates secrets handler for PUT method. It returns 405
"Method Not Allowed" for PUT method becuase secrets in docker swarm are
immutable.

Fixes: #65

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

#65 

## How Has This Been Tested?

I have tested this on Docker For Mac with Docker Swarm. For testing use `viveksyngh/faas-swarm:latest-dev` docker image.

```
faas-cli secret ls
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
No secrets found.
➜  faas-swarm git:(issue-65) ✗ faas-cli secret create test-secret --from-literal=test-value
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
Creating secret: test-secret
Created: 201 Created
➜  faas-swarm git:(issue-65) ✗ faas-cli secret ls
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.

NAME
test-secret

➜  faas-swarm git:(issue-65) ✗ faas-cli secret update test-secret --from-literal=update-value
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
Updating secret: test-secret
server returned unexpected status code: 405 - %                                                                                                                                   ➜  faas-swarm git:(issue-65) ✗ faas-cli secret rm test-secret
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
Removed.. OK.
➜  faas-swarm git:(issue-65) ✗ faas-cli secret ls
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
No secrets found.
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: handlers/secrets.go [hunk 1]
```diff
@@ -99,28 +95,28 @@ func getSecretsWithLabel(c client.SecretAPIClient, labelName string, labelValue
...
 				name,
 				ownerLabel,
 				ownerLabelValue,
-			), http.StatusInternalServerError
+			)
 		}
 	}
 
-	return nil, fmt.Errorf("not found secret with name: %s", name), http.StatusNotFound
+	return nil, http.StatusNotFound, fmt.Errorf("not found secret with name: %s", name)
```
Comment (alexellis) [inline handlers/secrets.go]: `unable to find secret with name: %s`

File: handlers/secrets.go [hunk 2]
```diff
@@ -51,7 +51,8 @@ func MakeSecretsHandler(c client.SecretAPIClient) http.HandlerFunc {
 			responseStatus, responseBody, responseErr = createNewSecret(c, body)
 			break
 		case http.MethodPut:
-			responseStatus, responseBody, responseErr = updateSecret(c, body)
+			responseStatus = http.StatusMethodNotAllowed
+			responseErr = fmt.Errorf("docker swarm secrets are immutable")
```
Comment (alexellis) [inline handlers/secrets.go]: `faas-swarm is unable to update secrets, delete and re-create or use a new name`

Comment (alexellis) [reply handlers/secrets.go]: Please could you add a unit test for this scenario if we don't have one yet?

Comment (viveksyngh) [reply handlers/secrets.go]: We already had unit test for this scenario, I have updated it.

**Pull request merged**